### PR TITLE
fix(frontend): keep MCP configuration dialog within viewport

### PR DESCRIPTION
**Pull request by AnnaSuSu**
## Why

Opening a long MCP server definition in Capability Center lets the JSON textarea grow beyond the viewport. The centered dialog then clips its title/close control and Save/Cancel buttons, making the configuration difficult to edit or dismiss on smaller windows.

## What changed

- Keep the shared add/edit MCP dialog within the viewport and give it more horizontal space on desktop.
- Scroll long JSON inside the editor. For exceptionally small viewports, long server names, or long validation messages, scroll the description/editor/error region while keeping the title, Close, Save, and Cancel outside that region.
- Cover editing and adding long definitions at 1280×720, 390×667, and 844×390, plus fallback scrolling at 500×280 and at 844×390 with a long server name. Exercise duplicate-name and malformed-JSON feedback, scrolling, Close, and Cancel.

The runtime change is limited to `frontend/src/components/workspace/capabilities/mcp-plugin-manager.tsx`; regression coverage lives in `frontend/tests/e2e/mcp-settings.spec.ts`.

## Surface area

- [x] **Frontend UI** — Capability Center MCP configuration dialog
- [ ] **Backend API**
- [ ] **Agents / LangGraph**
- [ ] **Sandbox**
- [ ] **Skills**
- [ ] **Dependencies**
- [ ] **Default behavior change**
- [ ] **Docs / tests / CI only**

## Screenshots / Recording

Entry point: **Workspace → Capability Center → Plugins → Edit an MCP plugin / Add MCP plugin**.

### Before

image

### After

image

## Bug fix verification

- Test: `frontend/tests/e2e/mcp-settings.spec.ts`, the three `long MCP definitions stay usable` cases.
- All three viewport checks failed before the layout fix: the dialog exceeded the viewport. The pre-fix MCP manager, shared Dialog, and Textarea files are byte-identical to `main` at `0efdf8e7`.
- The same checks passed after the fix, together with the existing configuration-save regression. Browser fixtures use synthetic server definitions and mocked backend APIs.

- Review fallback regressions: both new scenarios fail on the reviewed commit `953349f9` because scrolling moves the title out of view. They pass with the fixed outer title/actions and inner scrolling region. Both add and edit use the same layout; the tests also force long duplicate-name validation messages in add mode.

## Validation

- `pnpm format`: passed.
- `pnpm check` (ESLint and TypeScript): passed.
- Full frontend unit suite, `pnpm exec rstest run`: **200 files / 1,622 tests passed**.
- Focused Chromium checks against the development server: **6 passed**, including the two fallback cases, three long-JSON viewport cases, and the existing save flow.
- Production `pnpm build` with local development auth settings: passed.
- Complete Chromium suite against the production build, `pnpm exec playwright test --workers=1`: **237 passed**.
- `git diff --check`: clean.

## AI assistance

**Tool(s) used:** Codex

**How you used it:** Investigated the reported screenshot, implemented the scoped dialog layout fix and browser regressions, ran local validation, and prepared this PR. The author requested the fix and publication; the human code-review attestation below is left for the author.

- [x] I've read and understand every line of this change and take responsibility for it — it's not unreviewed AI output.

### CODE REVIEW

File: frontend/src/components/workspace/capabilities/mcp-plugin-manager.tsx [hunk 1]
```diff
@@ -287,8 +287,8 @@ function MCPServerList({
         open={editor !== null}
         onOpenChange={(open) => !open && !isWriting && closeEditor()}
       >
-        <DialogContent>
-          <DialogHeader>
+        <DialogContent className="flex max-h-[calc(100dvh-2rem)] flex-col overflow-y-auto sm:max-w-2xl">
```
Comment (willem-bd) [inline frontend/src/components/workspace/capabilities/mcp-plugin-manager.tsx]: The `shrink-0` classes keep the header, error, and footer from *shrinking*, but once the dialog's own `overflow-y-auto` fallback engages they still *scroll away with the content* — including the `absolute top-4 right-4` close button rendered inside this scroll container. I reproduced this with the exact dialog structure in headless Chromium: at a 500×280 window, or even at 844×390 once a long server name wraps inside the `editServerDescription` header (the name is user-configured and interpolated into `"{name}"`), the container overflows and scrolling down moves the Close button fully out of the viewport, while the footer is out of view until you scroll. None of the three new e2e viewports reach this path — with a short header the container only overflows below roughly 300px of viewport height — so the fallback stays untested. Not a blocker (still strictly better than the pre-fix clipping), but consider adding a short-viewport e2e case for the fallback, and/or keeping the close button (e.g. `sticky` or outside the scrollable region) reachable when it activates.

**Review by willem-bd — COMMENTED** [review]
Reviewed at 953349f (single commit, 2 files). The layout fix itself is sound: I verified with tailwind-merge that the added classes correctly override the shared Dialog/Textarea defaults (`grid`→`flex`, `sm:max-w-lg`→`sm:max-w-2xl`, `field-sizing-content`→`field-sizing-fixed`, `min-h-16`→`min-h-24`), and reproduced the dialog structure in headless Chromium to confirm the flex column shrinks the fixed-height textarea down to its `min-h-24` floor before the container ever overflows at 1280×720 and 844×390. The new e2e coverage follows the existing file's conventions (route precedence over `mockLangGraphAPI`, en-US role labels matching `en-US.ts`, per-test screenshots via `testInfo.outputPath`). One suggestion-level finding below about the container-scroll fallback that none of the new viewports exercise.

File: frontend/src/components/workspace/capabilities/mcp-plugin-manager.tsx (hunk 1 above)
Comment (AnnaSuSu) [reply frontend/src/components/workspace/capabilities/mcp-plugin-manager.tsx]: Confirmed both fallback cases on 953349f9 and fixed them in 4fcb03a1. The dialog shell no longer scrolls: the title/Close control and Save/Cancel footer stay outside an inner scroll region containing the description, JSON editor, and validation message. Long names and error text can therefore scroll without taking the controls with them.

Added regressions at 500×280 and at 844×390 with a long server name. Both fail on the reviewed commit and pass after the fix. They exercise the actual overflowing ancestor in both add/edit flows, including a long duplicate-name error, malformed JSON, and dismissing through Close/Cancel after scrolling. The existing three long-JSON viewport cases and save regression also pass.

Validation: pnpm format/check, production build, all 1,622 frontend unit tests, and all 237 production Chromium E2E tests passed. Updated the PR description and retained the uploaded screenshots.

Comment (willem-bd) [reply frontend/src/components/workspace/capabilities/mcp-plugin-manager.tsx]: Independently verified at 4fcb03a1: I ran the two new fallback-scrolling e2e cases (500x280, and 844x390 with a 510-char server name, edit + add flows) against a production build — all 6 mcp-settings tests pass with the fix, and both new cases fail on the reviewed commit 953349f9 with the dialog heading/Close at viewport ratio 0 after the fallback scroll (exact repro of the original finding). The title/Close control and Save/Cancel footer are now outside the inner `overflow-y-auto` region, so they stay pinned. Round-1 suggestion is resolved.

**Review by willem-bd — COMMENTED** [review]
Round 2 re-review at 4fcb03a1 (delta 953349f9..4fcb03a1, 1 commit: mcp-plugin-manager.tsx + new e2e cases). No new findings. The round-1 suggestion is resolved: the dialog shell no longer scrolls — DialogContent is `overflow-hidden` with the title/Close control (`shrink-0 pr-6` header) and Save/Cancel footer outside an inner `min-h-0 overflow-y-auto` region holding only the description, JSON editor, and validation message; moving DialogDescription out of the header keeps its `text-center sm:text-left` alignment, so nothing is lost. I independently reproduced the red/green evidence: against a production build with headless Chromium, all 6 mcp-settings e2e tests pass at head, while with the component reverted to 953349f9 both new fallback cases (500x280, and 844x390 with a 510-char server name, edit + add flows) fail exactly as reported in round 1 — the dialog heading/Close drop to viewport ratio 0 after the fallback scroll. The overflowing-ancestor probe in the new tests is a nice touch: it exercises the actual fallback path instead of assuming which element scrolls. Good to merge from my side.

**Review by WillemJiang — APPROVED** [review]
